xinference/model/llm/pytorch/yi_vl.py

```diff
@@ -229,25 +199,72 @@ def chat(
         t = Thread(target=self._model.generate, kwargs=generate_kwargs)
         t.start()
 
+        if stream:
+            it = self._generate_stream(streamer, stop_str)
+            return self._to_chat_completion_chunks(it)
+        else:
+            c = self._generate(streamer, stop_str)
+            return self._to_chat_completion(c)
+
+    def _generate(self, streamer, stop_str) -> Completion:
         generated_text = ""
         for new_text in streamer:
             generated_text += new_text
             if generated_text.endswith(stop_str):
                 generated_text = generated_text[: -len(stop_str)]
-        r = self._parse_text(generated_text)
```

Comment:
Why do you remove the function?

Reply:
yes，I found it is mainly used for gradio ui, but not for a general  purpose, I referenced from official code like: 

https://github.com/01-ai/Yi/blob/main/VL/openai_api.py

the parse_text is defined in web_demo:

https://github.com/01-ai/Yi/blob/main/VL/web_demo.py

Reply:
Will it effect the result from gradio?

Reply:
and the similar codes is also appeared in web_demo.py of qwen_vl, but not in xinference qwen_vl.py:

![image](https://github.com/xorbitsai/inference/assets/918617/547fa599-7f02-4015-84a4-bc66fb1e1ecb)


so I thought it's not neccessary for low-level code

Reply:
OK, maybe we should move the code into chat_interface anyway.

Reply:
> Will it effect the result from gradio?

I tried times, no effect to normal usage, but not sure for special usage, and also asked ChatGPT for analysing the purpose of _parse_for, it seems to process the code reference like markdown format output